src/modules/app/createNavigateToIntegreatUrl.js

```diff
@@ -4,32 +4,51 @@ import createNavigateToCategory from './createNavigateToCategory'
 import type { Dispatch } from 'redux'
 import type { NavigationScreenProp } from 'react-navigation'
 import createNavigateToEvent from './createNavigateToEvent'
+import type { SetContentLanguageActionType } from './StoreActionType'
 
 export default (dispatch: Dispatch<*>, navigation: NavigationScreenProp<*>) =>
-  (url: string, cityCode: string, language: string) => {
+  ({url, cityCode, language}: {|url: string, cityCode: string, language: string|}) => {
     const parts = url.split('/').filter(segment => segment)
     const pathnameParts = parts.splice(2)
+    const newCity = pathnameParts[0]
+    const newLanguage = pathnameParts[1]
     const pathname = pathnameParts.reduce((acc, part) => `${acc}/${part}`, '')
 
-    if (pathnameParts[0] === cityCode && (!pathnameParts[1] || pathnameParts[1] === language)) {
-      // same city and language
+    // todo add support for linking to different cities
+
+    if (newCity === cityCode) {
+      // same city
+
+      if (newLanguage !== language) {
+        const setContentLanguage: SetContentLanguageActionType = {
+          type: 'SET_CONTENT_LANGUAGE',
+          params: {
+            newLanguage
+          }
+        }
+        dispatch(setContentLanguage)
```

Comment:
I'm concerned of the `state.cityContent.language`now: I think it's not at all clear, what this value holds now: Does it hold the "current language of the screen"?
Also what happens with it when navigating back (and potentially changing the "current language of the screen")? I think it would be a lot easier to just switch the content-language (as the SWITCH_CONTENT_LANGUAGE action does) and then push the new page on top... (so the whole stack has the same language)

Edit: I saw now, that you handled this "moving-back"-case. Still, this somehow bugs me...

Reply:
I don't really like this approach. If a category/event in a different language is linked and the user would navigate back, the previous category would also be in the new language, which isn't expected by users imo.

Reply:
Yes, that's true; I think as a user, it makes more sense to have the original language when navigating back... On the other hand this increases complexity a lot (having multiple languages opened at the same time) for a use-case that should not occur (because the links deployed by the cms should target the same language)... :thinking:

Reply:
So why do we even need this? This task is solely about linking to stuff in a different language. If there are only links in the same language, this is not needed at all.

Reply:
We decided to do this because we did not really know what we should do in case of a link like: `/ausgburg/de/gesundheit` -> `/ausgburg/en/first-steps`.

I'm very sure that that navigating back should always yield to the page you previously saw. So if a city decides to link to a page in an other language, e.g. de -> ar, then navigating back should bring you back to the de page.

There is actually a usecase for this if a page only exists in one particular language.

TL;DR: We should not ignore the language in a link. Navigating back should yield the previous page and therefore change the language.

I'm not deep into the code, but that is the way is is solved right now, right?

Reply:
Well, the problem is, there are links targeting different languages. Although this is a problem of our cms, we need to somehow react adequately to that.

Reply:
@maxammann Alright, but then we should consider removing the language as a property in the `state.cityContent.language`, since there's no real concept behind that, and instead just have a language per screen...

Reply:
@Schedulaar Most links are probably correct in the CMS. Handling it the way I described is at least the "right" way to do it. So yes you are right language is an attribute of the current page.

BUT: I could live if it is like @Schedulaar described in the first post for now.


@klinzo It's up to you how you want to solve it:

1. Make language an attribute of the current route
2. Just switch the content-language (as the SWITCH_CONTENT_LANGUAGE action does) and then push the new page on top... (so the whole stack has the same language)

I think the concept of a previousLanguage is not very clean and you probably have an issue in the following case: You follow the links: `/augsburg/de/` -> `/augsburg/en/` -> `/augsburg/ar/`.

If you navigate back now 2 times the "previousLanguage" de is lost.

Reply:
@maxammann No, the previous language would not be lost if navigating back 2 times.

Reply:
I think we should discuss this in person.

Reply:
Alright, Tuesday?

Reply:
So I thought about this once again:
We also need to consider what happens when the user follows a link to a different language and then explicitly switches the language on that page (Also: Which language should the modal display as `selected`? I think the current language of the page).
Imo the following concept would make sense if we want to support this navigating-back-preserves-language feature:

We maintain the `state.cityContent.language´ not as the current-screen-language, but the "global" setting of the current contentLanguage (so it does not change when following a link to a different language). When changing a language explicitly by the user, this global language will be switched.
Other than that we should maintain the current language per route/screen. As far as I can see, we would not need to remember this `previousLanguage` that way.
Using this, we need to somehow give the `LanguageModal` the current language of the screen, which should technically be possible afaik.

Reply:
I think @Schedulaar s post makes a lot of sense. Ofc this is a complicated solution but behaves like I would expect.

Reply:
So if the language is switched explicitly by the user, all route language props are switched to the newLanugae?